UNIX domain socket support for GSB

Wiezzel opened the pull request.

Wiezzel committed:
UNIX domain socket support for GSB

Signed-off-by: Adam Wierzbicki <awierzbicki@golem.network>

mfranciszkiewicz commented on crates/router/src/lib.rs:
```diff
@@ -9,10 +9,12 @@ use std::fmt::{Debug, Display};
 use std::net::SocketAddr;
 use std::sync::Arc;
-use tokio::net::{TcpListener, TcpStream};
+use tokio::net::{TcpListener, TcpStream, UnixListener, UnixStream};
```
Not platform independent

Wiezzel committed:
UNIX socket support enabled only on UNIX systems

maaktweluit reviewed (approved):
LGTM!

Wiezzel committed:
DEFAULT_GSB_URL made OS-dependent

For UNIX systems (which support UNIX domain sockets) the default is
"unix:/tmp/yagna.sock" because it provides more security. For non-UNIX
systems fall back to the old value "tcp://127.0.0.1:7464".

Signed-off-by: Adam Wierzbicki <awierzbicki@golem.network>

Wiezzel commented:
@mfranciszkiewicz Issue with ya-provider resolved by setting `DEFAULT_GSB_URL` to `unix:/tmp/yagna.sock` on UNIX.

Wiezzel commented:
@maaktweluit Bump

maaktweluit reviewed (approved):
Had reviewed this already last thursday :S 

LGTM!

Should this go to `payments/beta-1` or `release/v0.6` to keep in sync with yagna?
